apps/judicial-system/web/src/utils/validate.ts

```diff
@@ -298,8 +298,14 @@ export const isTrafficViolationStepValidIndictments = (
   workingCase: Case,
 ): boolean => {
   return Boolean(
-    workingCase.demands &&
-      (!workingCase.hasCivilClaims || workingCase.civilDemands),
+    hasOnlyOneItemInSubArrays(workingCase.indictmentSubtypes) ||
+      (workingCase.indictmentCounts?.every(
+        (indictmentCount) =>
+          indictmentCount.indictmentCountSubtypes &&
+          indictmentCount.indictmentCountSubtypes?.length > 0,
+      ) &&
+        workingCase.demands &&
```

Comment:
Did you mean to stop doing the workingCase.demands checks if hasOnlyOneItemInSubArrays returns true?

Reply:
Yes, and that is clearer now that I've refactored the code. Please take another look and report back if anything is unclear :)